lib/src/local_working_copy.rs

```diff
@@ -1168,8 +1170,27 @@ impl TreeState {
                     }))
                 }
                 Err(new_file_ids) => {
-                    if new_file_ids != old_file_ids {
-                        Ok(current_tree_values.with_new_file_ids(&new_file_ids))
+                    if new_file_ids != old_file_ids && new_file_ids != simplified_old_file_ids {
+                        if old_file_ids == simplified_old_file_ids {
+                            Ok(current_tree_values.with_new_file_ids(&new_file_ids))
+                        }
+                        // If the conflict was simplified, expand the conflict to the original
+                        // number of sides.
+                        else {
+                            let new_file_ids = new_file_ids.iter().collect_vec();
+                            let new_file_ids = old_file_ids
+                                .iter()
+                                .map(|file_id| {
+                                    let index = simplified_old_file_ids
+                                        .iter()
+                                        .position(|old_file_id| old_file_id == file_id)
+                                        .unwrap();
+                                    new_file_ids[index].clone()
+                                })
+                                .collect_vec();
+                            Ok(current_tree_values
+                                .with_new_file_ids(&Merge::from_vec(new_file_ids)))
+                        }
```

Comment:
Should this code move into `update_from_content()`? That would make it work for our custom working copy implementation at Google. It's also related to the question above: by giving `update_from_content()` the unsimplified conflict, maybe we can make that function able to also accept parsing the unsimplified conflicts, perhaps by double-checking after a conflict appears to be resolved that it can't be parsed as the unsimplified conflict.

Reply:
Makes sense, will shift this over.